pkg/build/util/generate.go

```diff
@@ -13,6 +14,7 @@ import (
 // GenerateBuildFromConfig creates a new build based on a given BuildConfig. Optionally a SourceRevision for the new
 // build can be specified.  Also optionally a list of image names to be substituted can be supplied.  Values in the BuildConfig
 // that have a substitution provided will be replaced in the resulting Build
+// You NEED to update BuildConfig object after calling this!
```

Comment:
technically the same issue exists when we update the lasttriggeredid on the buildconfig in the imagechange_controller.  Not having transactions is awesome.

The way around it would be to break the sequenceid out of the builconfig object and have it be its own etcd key.